rubrik_cdm/data_management.py

```diff
@@ -659,12 +680,101 @@ def assign_sla(self, object_name, sla_name, object_type, log_backup_frequency_in
+                    if log_retention_hours is not None:
+                        config['logRetentionHours'] = log_retention_hours
+                    if num_channels is not None:
+                        config['numChannels'] = num_channels
+
+                    config['configuredSlaDomainId'] = sla_id
+
+                    patch_resp = self.patch("internal", "/oracle/db/{}".format(oracle_db_id), config, timeout)
+            else:
+                raise InvalidParameterException(
+                    "Database ID not found for instance '{}'").format(object_name)
+            
+            return patch_resp
+
+        elif object_type == 'oracle_host':
+
+            host_id = ''
+            db_sla_lst = []
+
+            self.log('Searching the Rubrik cluster for the current Oracle hosts.')
+            host_id = self.object_id(object_name, object_type)
+
+            if(host_id):
+                self.log(
+                    "assign_sla: Determing the SLA Domain currently assigned to the Oracle Host '{}'.".format(object_name))
+
+                oracle_summary = self.get(
+                    'internal',
+                    '/oracle/host/{}'.format(host_id),
+                    timeout=timeout)
+
+                if (sla_id == oracle_summary['configuredSlaDomainId'] and log_backup_frequency_in_minutes == oracle_summary['logBackupFrequencyInMinutes'] and
+                        log_retention_hours == oracle_summary['logRetentionHours'] and num_channels == oracle_summary['numChannels']):
+                    return "No change required. The Oracle Host '{}' is already assigned to the '{}' SLA Domain with the following log settings:" \
+                            " log_backup_frequency_in_minutes: {}, log_retention_hours: {} and num_channels: {}.".format(
+                                object_name, sla_name, log_backup_frequency_in_minutes, log_retention_hours, num_channels)
+
+                else:
+                    self.log(
+                        "assign_sla: Assigning the Oracle Host '{}' to the '{}' SLA Domain.".format(
+                            object_name, sla_name))
+
+                    config = {}
+                    if log_backup_frequency_in_minutes is not None:
+                        config['logBackupFrequencyInMinutes'] = log_backup_frequency_in_minutes
+                    if log_retention_hours is not None:
+                        config['logRetentionHours'] = log_retention_hours
+                    if num_channels is not None:
+                        config['numChannels'] = num_channels
+
+                    config['configuredSlaDomainId'] = sla_id
+
+                    patch_resp = self.patch("internal", "/oracle/host/{}".format(host_id), config, timeout)
+                    db_sla_lst.append(patch_resp)
+            else:
+                raise InvalidParameterException(
+                    "Host ID not found for instance '{}'").format(object_name)
+
+            return db_sla_lst            
```

Comment:
@shamsway down here too. Sorry, should have mentioned that.

Reply:
@drew-russell Well, I knew that, and I thought I'd fixed it, but I was in the wrong object type 😬 . I've undone the unintended changes, and fixed it in the `oracle_host` object type code block. Sheesh.